husarion_ugv_crsf_teleop/husarion_ugv_crsf_teleop/crsf_teleop_node.py

```diff
@@ -0,0 +1,187 @@
+            silence_cmd_vel = self.get_parameter('enable_cmd_vel_silence_switch').get_parameter_value().bool_value
+            
+            if send_cmd_vel and not silence_cmd_vel:
+                linear_speed_modifier = self.linear_speed_presets[round(channels[10]) + 1]
+                angular_speed_modifier = self.angular_speed_presets[round(channels[10]) + 1]
+                
+                t = Twist()
+                t.linear.x = channels[1] * linear_speed_modifier
+                t.angular.z = -channels[3] * angular_speed_modifier
+                
+                self.cmd_vel_publisher.publish(t)
+        
+        elif msg.msg_type == PacketType.LINK_STATISTICS:
+            last_lq = self.link_status.lq
+            
+            self.link_status.rssi_1 = -msg.payload[0]
+            self.link_status.rssi_2 = -msg.payload[1]
+            self.link_status.lq = msg.payload[2]
+            self.link_status.uplink_snr = int.from_bytes(bytes(msg.payload[3]), byteorder='big', signed=True)
+            self.link_status.used_antenna = msg.payload[4]
+            self.link_status.mode = msg.payload[5]
+            self.link_status.tx_power = msg.payload[6]
+            self.link_status.downlink_rssi = -msg.payload[7]
+            self.link_status.downlink_lq = msg.payload[8]
+            self.link_status.downlink_snr = int.from_bytes(bytes(msg.payload[9]), byteorder='big', signed=True)
+            
+            # Detect connection status
+            if last_lq != 0 and self.link_status.lq == 0:
+                self.get_logger().error("Connection lost")
+                
+                # Publish empty cmd_vel to stop the robot
+                self.cmd_vel_publisher.publish(Twist())
+                
+            if last_lq == 0 and self.link_status.lq > 0:
+                self.get_logger().info("Connected")
+            
+            # Warn on low link quality
+            if last_lq != self.link_status.lq:
+                if self.link_status.lq < 15:
+                    self.get_logger().warn(f"Very low link quality: {self.link_status.lq}%")
+                elif self.link_status.lq < 30:
+                    self.get_logger().warn(f"Low link quality: {self.link_status.lq}%")
+                elif last_lq < 30 and self.link_status.lq >= 30:
+                    self.get_logger().info(f"Link quality restored: {self.link_status.lq}%")
+            
+            self.link_status_publisher.publish(self.link_status)
+            
+        else:
+            self.get_logger().warn(f"Unknown CRSF message (Type: {msg.msg_type.name}, Length: {msg.length})")
+            
+    def update_e_stop(self):
+        if self.rc_estop_state:
+            self.e_stop_trigger.call_async(Trigger.Request())
+
+
+def main(args=None):
+    rclpy.init(args=args)
+
+    crsf_interface = CRSFInterface()
+    
+    rclpy.spin(crsf_interface)
+    
+    crsf_interface.destroy_node()
+    rclpy.shutdown()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
format with precommit

Reply:
Does the setup.py need copyright notice? I don't think so, but pre-commit hook thinks otherwise

Reply:
I check the others and they add copyright to setup.py ;)

https://github.com/ros-controls/ros2_controllers/blob/master/rqt_joint_trajectory_controller/setup.py

Reply:
Then please add precommit configuration and precommit workflow